MoreLinq/GroupPartition.cs

```diff
@@ -0,0 +1,144 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2009 Atif Aziz. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Partitions a grouping by Boolean keys into a projection of true
+        /// elements and false elements, respectively.
+        /// </summary>
+
+        public static TResult Partition<T, TResult>(this IEnumerable<IGrouping<bool, T>> source,
+            Func<IEnumerable<T>, IEnumerable<T>, TResult> resultSelector) =>
+            source.Partition(true, false, (t, f, _) => resultSelector(t, f));
+
+        /// <summary>
+        /// Partitions a grouping by nullable Boolean keys into a projection of
+        /// true elements, false elements and null elements, respectively.
+        /// </summary>
+
+        public static TResult Partition<T, TResult>(this IEnumerable<IGrouping<bool?, T>> source,
+            Func<IEnumerable<T>, IEnumerable<T>, IEnumerable<T>, TResult> resultSelector) =>
+            source.Partition(true, false, null, (t, f, n, _) => resultSelector(t, f, n));
```

Comment:
I do that where it's limited to one unused parameter as that's all C# can afford us. Do you suggest to use 2 underscores for the second unused? As in:

```c#
PartitionImpl(source, 1, key, default(TKey), default(TKey), comparer, (a, _, __, rest) => resultSelector(a, rest));
```